File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int>mp;
        int l{0},r{0};
        int ans{0};
        while (r<s.length()){
            mp[s[r]]++;
            while (mp[s[r]]>1){
                mp[s[l]]--;
                l++;
            }
            ans = max(ans,r-l+1);
            r++;
        }
        return ans;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        temp = set()
        l,ans = 0,0
        for r in range(len(s)):
            if s[r] not in temp:
                temp.add(s[r])
                ans = max(ans,r-l+1)
            else:
                while s[r] in temp:
                    temp.remove(s[l])
                    l +=1
                temp.add(s[r])
        return ans